open-craft/frontend-app-discussions
# feat: update MFE to use dynamically injected theme

## Description

Updates the MFE to use alpha version of frontend build, platform and paragon to support runtime theming.

## Supporting information

https://openedx.atlassian.net/wiki/spaces/BPL/pages/3770744958/Migrating+MFEs+to+Paragon+design+tokens+and+CSS+variables

## Testing instructions

1. Clone https://github.com/open-craft/edx-simple-theme and check out branch `artur/design-tokens-update`.
2. Build it (`npm ci && npm run build`).
3. Run a local server from the `dist` directory (e.g. `python3 -m http.server 9100`, use same port for `brandOverride` in MFE_CONFIG below).
4. Set MFE_CONFIG as in example Tutor plugin below.
5. If using tutor, enable the forum with `tutor plugins enable forum`.
6. Go to a course and select the "Discussion" tab.
7. See that theming works.
8. Check that `http://localhost:9100/light.min.css` was loaded.

<details>
<summary>Example Tutor plugin to set MFE_CONFIG</summary>

```
name: mfe_config
version: 0.1.0
patches:
  mfe-lms-common-settings: |
    MFE_CONFIG.update({
        "PARAGON_THEME_URLS": {
            "core": {
                "urls": {
                    "default": "https://cdn.jsdelivr.net/npm/@openedx/paragon@23.0.0-alpha.1/dist/core.min.css",
                    "brandOverride": "https://cdn.jsdelivr.net/npm/@edx/brand-edx.org@2.2.0-alpha.17/dist/core.min.css"
                }
            },
            "variants": {
                "light": {
                    "urls": {
                        "default": "https://cdn.jsdelivr.net/npm/@openedx/paragon@23.0.0-alpha.1/dist/light.min.css",
                        "brandOverride": "http://localhost:9100/light.min.css"
                    },
                    "default": True,
                    "dark": False
                }
            }
        }
    })
```
</details>

## Deadline

ASAP

## Other information

Internal-ref: https://tasks.opencraft.com/browse/BB-9023
